pwiz_tools/Skyline/SkylineFiles.cs

```diff
@@ -2601,6 +2638,39 @@ private static IEnumerable<Target> CheckMissingIrtPeptides(SrmDocument document)
                 .Where(target => !existingPeptides.ItemsMatching(new LibKey(target, Adduct.EMPTY).LibraryKey, false).Any());
         }
 
+        public static bool CheckDecoys(SrmDocument document, out int numDecoys, out int numNoSource, out int numWrongTransitionCount)
+        {
+            var targets = new Dictionary<Target, List<PeptideDocNode>>();

```

Comment:
I think you should do:
targets = document.Peptides.Where(pep=>!pep.IsDecoy).ToLookup(pep=>pep.ModifiedTarget)

Reply:
You are right, this is a great spot for a lookup.